### PR TITLE
Fix: Prettifying unexpected error on FileExists for get/get-url/imp-url/add and introducing --force

### LINKED ISSUE #5387: get: unexpected error, if the file to download already exists
# Bug Report
```console
$ dvc get https://github.com/iterative/dvc README.rst -o reamde2.rst
$ dvc get https://github.com/iterative/dvc README.rst -o reamde2.rst -v
2021-02-02 16:10:25,831 ERROR: unexpected error - [Errno 17] File exists: 'reamde2.rst'
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/saugat/repos/iterative/dvc/dvc/main.py", line 50, in main
    ret = cmd.run()
  File "/home/saugat/repos/iterative/dvc/dvc/command/get.py", line 31, in run
    return self._get_file_from_repo()
  File "/home/saugat/repos/iterative/dvc/dvc/command/get.py", line 37, in _get_file_from_repo
    Repo.get(
  File "/home/saugat/repos/iterative/dvc/dvc/repo/get.py", line 26, in get
    out = resolve_output(path, out)
  File "/home/saugat/repos/iterative/dvc/dvc/utils/__init__.py", line 366, in resolve_output
    raise FileExistsError(errno.EEXIST, os.strerror(errno.EEXIST), ret)
FileExistsError: [Errno 17] File exists: 'reamde2.rst'
------------------------------------------------------------
2021-02-02 16:10:26,617 DEBUG: Version info for developers:
DVC version: 2.0.0a0+d05be3.mod 
---------------------------------
Platform: Python 3.9.1 on Linux-5.10.11-arch1-1-x86_64-with-glibc2.32
Supports: azure, gdrive, gs, http, https, s3, ssh, oss, webdav, webdavs, webhdfs
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sda9
Caches: local
Remotes: https
Workspace directory: ext4 on /dev/sda9
Repo: dvc, git

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
2021-02-02 16:10:26,619 DEBUG: Analytics is disabled.
```

### PR DESCRIPTION
**Pull request by omesser**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Closes #5387

### Changes:
- [x]  Fixing error output on dvc get/get-url/import-url/add upon conflict with a local file

  Example:
  ```sh
  ❯ python -m dvc get https://github.com/iterative/dvc README.rst -o test.rst
  ❯ python -m dvc get https://github.com/iterative/dvc README.rst -o test.rst
  ERROR: failed to get 'README.rst' from 'https://github.com/iterative/dvc' - The file 'test.rst' already exists locally. 
  To override it, re-run with '--force'.
  ❯ python -m dvc get https://github.com/iterative/dvc README.rst -o test.rst --force
  ```
- [x] Adding `--force` flag to override
- [x] tests
- [ ] updating docs with `--force` <- not done yet

### CODE REVIEW

File: dvc/exceptions.py [hunk 1]
```diff
@@ -206,6 +207,18 @@ def __init__(self, etag, cached_etag):
...
+class FileExistsLocallyError(FileExistsError, DvcException):
+    def __init__(self, path, hint=None):
+        import os.path
+
+        self.path = path
+        hint = "" if hint is None else f". {hint}"
+        path_typ = "directory" if os.path.isdir(path) else "file"
+        msg = f"The {path_typ} '{path}' already exists locally{hint}"
+        super().__init__(msg)
+        self.errno = errno.EEXIST
```
Comment (omesser) [inline dvc/exceptions.py:219]: Setting errno after initializing super to keep the error string nice (no need for "[Errno 17]")
Note that I kept inheritance from `FileExistsError` and the errno for canonical behavior and to maintain backwards compatibility of the api in case people use this to catch those errors. I'm not sure it's needed or important, though 🤷

File: dvc/repo/get.py [hunk 2]
```diff
@@ -20,14 +20,14 @@ def __init__(self):
...
-def get(url, path, out=None, rev=None, jobs=None):
+def get(url, path, out=None, rev=None, jobs=None, force=False):
     import shortuuid
 
     from dvc.dvcfile import is_valid_filename
     from dvc.external_repo import external_repo
     from dvc.fs.callbacks import Callback
 
-    out = resolve_output(path, out)
+    out = resolve_output(path, out, override=force, override_hint=True)
```
Comment (omesser) [inline dvc/repo/get.py:30]: override vs force

Would be more natural for me to add an `--override` flag, but I went with force because:
- I think sticking to this "generic name" is more consistent with naming style in DVC
- was @dberenbaum 's suggestion in the issues [comment](https://github.com/iterative/dvc/issues/5387#issuecomment-1048026717)
- intuitive short form `-f` (`-o` taken)

Comment (daavoo) [reply dvc/repo/get.py:30]: No strong opinion, but I would also name the internal arguments `force`, I think it is not only the DVC naming style but very common in most CLI tools (https://clig.dev/#arguments-and-flags)

Comment (daavoo) [reply dvc/repo/get.py:30]: In any case, it should be `overwrite` instead of `override`, right?

File: dvc/exceptions.py [hunk 3]
```diff
@@ -206,6 +207,18 @@ def __init__(self, etag, cached_etag):
         )
 
 
+class FileExistsLocallyError(FileExistsError, DvcException):
+    def __init__(self, path, hint=None):
```
Comment (daavoo) [inline dvc/exceptions.py:211]: Do we really need to make the hint conditional?
What is the case where we don't want the hint?

**Review by daavoo — APPROVED** [review]
Minor comments but LGTM.

Not meaning to expand the scope of the P.R. but I think the only usage left of `resolve_output` is in `dvc add -o`.
I believe that it would also make sense to add support for `--force` there.

File: dvc/repo/get.py (hunk 2 above)
Comment (omesser) [reply dvc/repo/get.py]: 👍  I've added support to `dvc add [target] -o [file] --force` with consistent behavior.

File: dvc/exceptions.py (hunk 3 above)
Comment (omesser) [reply dvc/exceptions.py:211]: 👍  I've added support to `dvc add [target] -o [file] --force` with consistent behavior.
Now the hint is not conditional

File: dvc/commands/add.py [hunk 4]
```diff
@@ -140,6 +141,14 @@ def add_parser(subparsers, parent_parser):
         ),
         metavar="<number>",
     )
+    parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        default=False,
+        help="Ignores overwrites to dvc.yaml file and allows overwriting local stage"
```
Comment (daavoo) [inline dvc/commands/add.py:149]: What do you mean by `overwrites to dvc.yaml` here? When does `dvc add` affect `dvc.yaml`?

Comment (omesser) [reply dvc/commands/add.py]: ugh, my bad.
tl;dr - you're right and it doesn't. I'm changing this help message to be the same as others

what threw me off here is that add command actually creates single-stages for all targets and uses the stage objects to check the graph (🤷‍♂️ I guess shoving the targets into stages is a way to conform to updating the index which happens there).
So, when creating the stage objects, `force` is actually passed there, BUT, upon re-checking this logic, it's ineffectual in this flow (because it's only inspected if stage is created with `validate=True` to opt out of some validation 😄 
So, to summarize - I was mistaken, the force flag ONLY has the effect added in this PR to override local files/folders if `--out <PATH>` is specified

The end

**Comment by dberenbaum** [comment]
@omesser I just ran into this error:

```bash
$ dvc import git@github.com:dberenbaum/lstm_seq2seq.git model
ERROR: failed to import 'model' from 'git@github.com:dberenbaum/lstm_seq2seq.git'. - The directory 'model' already exists locally.
To override it, re-run with '--force'.
```

This PR adds force but not for `dvc import`. Before I open another PR, any reason you didn't add `--force` to `dvc import`?
